x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/frequentitemsets/FrequentItemSetsAggregationBuilder.java

```diff
@@ -39,10 +44,12 @@ public final class FrequentItemSetsAggregationBuilder extends AbstractAggregatio
     public static final double DEFAULT_MINIMUM_SUPPORT = 0.01;
     public static final int DEFAULT_MINIMUM_SET_SIZE = 1;
     public static final int DEFAULT_SIZE = 10;
+    public static final Set<String> EXECUTION_HINT_ALLOWED_MODES = new TreeSet<>(Set.of("global_ordinals", "map"));
```

Comment:
Converting this to a `TreeSet` makes it mutable, which contradicts the all capitals variable name.

Since it's only got two elements I'd be inclined to just use `List.of` instead - it wouldn't be much less efficient for 2 items. Or otherwise just use `Set.of` and not worry about the order, which only matters in an error message.